Code/StructEstimation.py

```diff
@@ -124,20 +121,27 @@ def simBirth(self, which_agents):
         return None
 
 
-# Make a lifecycle consumer to be used for estimation, including simulated shocks (plus an initial distribution of wealth)
-# Make a TempConsumerType for estimation
-EstimationAgent = TempConsumerType(**Params.init_consumer_objects)
-# Set the number of periods to simulate
-EstimationAgent.T_sim = EstimationAgent.T_cycle + 1
-# Choose to track bank balances as wealth
-EstimationAgent.track_vars = ["bNrm"]
-# Draw initial assets for each consumer
-EstimationAgent.aNrmInit = DiscreteDistribution(
-    Params.initial_wealth_income_ratio_probs,
-    Params.initial_wealth_income_ratio_vals,
-    seed=Params.seed,
-).draw(N=Params.num_agents)
-EstimationAgent.make_shock_history()
+class IndShkLifeCycleConsumerType(TempConsumerType, IndShockConsumerType):
+    """
+    A very lightly edited version of IndShockConsumerType.  Uses an alternate method of making new
+    consumers and specifies DiscFac as being age-dependent.  Called "temp" because only used here.
+    """
+
+
+class PortfolioLifeCycleConsumerType(TempConsumerType, PortfolioConsumerType):
+    """
+    A very lightly edited version of PortfolioConsumerType.  Uses an alternate method of making new
+    consumers and specifies DiscFac as being age-dependent.  Called "temp" because only used here.
+    """
+
+    def check_restrictions(self):
+        return None
+
+    def post_solve(self):
+        for solution in self.solution:
+            solution.cFunc = solution.cFuncAdj
+            share = solution.ShareFuncAdj
+            solution.ShareFuncAdj = lambda m: np.clip(share(m), 0.0, 1.0)
```

Comment:
Fixed my issue with a kind of hack. This ensures that the share is always between 0 and 1. However, the right place to fix this is probably at the HARK code such that this hack functionally doesn't do anything.